pkg/scheduler/internal/cache/node_tree_test.go

```diff
@@ -406,39 +410,39 @@ func TestNodeTreeMultiOperations(t *testing.T) {
 		expectedOutput []string
 	}{
 		{
-			name:           "add and remove all nodes between two Next operations",
+			name:           "add and remove all nodes",
 			nodesToAdd:     allNodes[2:9],
 			nodesToRemove:  allNodes[2:9],
-			operations:     []string{"add", "add", "next", "add", "remove", "remove", "remove", "next"},
-			expectedOutput: []string{"node-2", ""},
+			operations:     []string{"add", "add", "add", "remove", "remove", "remove"},
+			expectedOutput: nil,
 		},
 		{
-			name:           "add and remove some nodes between two Next operations",
+			name:           "add and remove some nodes",
 			nodesToAdd:     allNodes[2:9],
 			nodesToRemove:  allNodes[2:9],
-			operations:     []string{"add", "add", "next", "add", "remove", "remove", "next"},
-			expectedOutput: []string{"node-2", "node-4"},
+			operations:     []string{"add", "add", "add", "remove"},
+			expectedOutput: []string{"node-3", "node-4"},
 		},
 		{
-			name:           "remove nodes already iterated on and add new nodes",
+			name:           "remove three nodes",
 			nodesToAdd:     allNodes[2:9],
 			nodesToRemove:  allNodes[2:9],
-			operations:     []string{"add", "add", "next", "next", "add", "remove", "remove", "next"},
-			expectedOutput: []string{"node-2", "node-3", "node-4"},
+			operations:     []string{"add", "add", "add", "remove", "remove", "remove", "add"},
+			expectedOutput: []string{"node-6"},
 		},
 		{
-			name:           "add more nodes to an exhausted zone",
+			name:           "6 nodes",
 			nodesToAdd:     append(allNodes[4:9], allNodes[3]),
 			nodesToRemove:  nil,
-			operations:     []string{"add", "add", "add", "add", "add", "next", "next", "next", "next", "add", "next", "next", "next"},
-			expectedOutput: []string{"node-4", "node-6", "node-7", "node-8", "node-3", "node-4", "node-6"},
+			operations:     []string{"add", "add", "add", "add", "add", "add"},
+			expectedOutput: []string{"node-4", "node-6", "node-3", "node-7", "node-8"},
 		},
 		{
-			name:           "remove zone and add new to ensure exhausted is reset correctly",
+			name:           "remove zone and add new",
 			nodesToAdd:     append(allNodes[3:5], allNodes[6:8]...),
```

Comment:
I found this is a wrong usage for the append. It will modify the variable 'allNodes' directly. After this line, the 'node-5' is removed from the 'allNodes'. We should copy the data from allNodes, not modify the allNodes directly. I submit a patch for it https://github.com/kubernetes/kubernetes/pull/93516.

Reply:
thanks. The two PRs will conflict though. Should we first merge this one and fix the conflict / rebase yours on top, or reverse ?

Reply:
@Maelk, np! I'm ok with both way, I'm a little prefer to merge https://github.com/kubernetes/kubernetes/pull/93516 first, since it makes this patch's unittests are more clear for the reviewers. But I'm really ok for any way, just let us see what is the preference from the reviewers.

Reply:
Note that we are in freeze. But I'm also inclined to merge #93516 first.

Reply:
ok, let's proceed with the other one first then.